### PR TITLE
Split distribution and spreading methods into separate methods for each axis

### PR DESCRIPTION
**Pull request by NickEntin**
@efirestone and I were working on some integrations and decided that these methods read a lot better when there are separate methods for vertical/horizontal distribution/spreading. Functionally these behave the same, but are syntactically easier to understand. This also means that we can differentiate the orthogonal alignment case names (e.g. use `.top` for vertical instead of `.leading` for everything).

Easiest to review by commit.

### CODE REVIEW

File: Paralayout/UIView+Spreading.swift [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        spreadOutSubviews(
+            subviews,
+            axis: .horizontal,
+            margin: margin,
+            inRect: bounds
+        ) { subviewFrame, subviewBounds, layoutBounds in
+            switch orthogonalBehavior {
+            case .fill:
+                // No-op. The `subviewFrame` has already been sized and positioned to fill the available vertical
+                // space in the container.
```
Comment (AliceAponasko) [inline Paralayout/UIView+Spreading.swift:101]: Thank you for the clarification! Really helpful.

File: Paralayout/ViewDistributionItem.swift [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+    /// Filter invisible views (`nil`, uninstalled, hidden, or transparent) from a distribution, and collapse adjacent
+    /// spacers (preferring larger ones).
+    ///
+    /// - parameter distribution: An array of optional distribution specifiers to be collapsed.
+    /// - returns: An array of `DistributionItem`s, without any invisible views or sequential `.fixed` or `.flexible`
+    /// spacers.
+    public static func collapsing(_ distribution: [ViewDistributionSpecifying?]) -> [ViewDistributionItem] {
+        var collapsedItems = [ViewDistributionItem]()
+
+        func isViewVisible(_ view: UIView) -> Bool {
```
Comment (AliceAponasko) [inline Paralayout/ViewDistributionItem.swift:42]: [nit] This can be a private extension on UIView.

**Review by AliceAponasko — APPROVED** [review]
Love this change! 🚢

File: Paralayout/ViewDistributionItem.swift (hunk 2 above)
Comment (NickEntin) [reply Paralayout/ViewDistributionItem.swift:42]: The behavior of this entire method is a bit confusing, so I think we're going to change it up. Still an open discussion around exactly how it should work though. See #29.

**Pull request merged**